### PR TITLE
[IMP] sale_order_product_recommendation: favorites, categories

### PR DESCRIPTION
**Pull request by yajo**
New shiny feature: you can un/favorite products, and those will appear first in the list of recommendations.

imagen

New not-so-shiny feature: recommendations will be sortable by product category.

imagen

While testing the module in the field, it turned out that salespeople often use recommendations based on the product category. Other than that, the favorites option still helps.

@moduon MT-4472

### CODE REVIEW

**Comment by sergio-teruel** [comment]
You are breaking the module behavior.

Add a setting option to manage this!!

**Comment by sergio-teruel** [comment]
Now you always order by category first.... I know this problem... 
the categ orderd is incompatible with the more delivered order....

File: sale_order_product_recommendation/wizards/sale_order_recommendation.py
Comment (yajo) [inline sale_order_product_recommendation/wizards/sale_order_recommendation.py]: Before this patch, it was just ordered by id. That default order didn't seem very meaningful, that's why I changed it so freely, but I didn't mean to break the module.

I still think it makes sense to put first the favorites, isn't it?

So, the correct default order would be this?

```suggestion
    _order = "product_categ_id, units_delivered desc, times_delivered desc, id"
```

Or would it be just this other one?

```suggestion
    _order = "product_priority desc, id"
```

Comment (yajo) [reply sale_order_product_recommendation/wizards/sale_order_recommendation.py]: I decided to go with the 2nd option, which is more behavior-preserving.

That way, this improvement can be merged faster. Still, on desktop view, users can click on the category header to be able to sort by category.

We will see which other options we have for implementing that order by default later.

**Comment by rafaelbn** [comment]
Hello @sergio-teruel ! Your comments have beed addressed 😄 Do you agree?

**Comment by rafaelbn** [comment]
@yajo ,

> We will see which other options we have for implementing that order by default later.

Diferent companies have diferent default order criteria, so by parameter could be good options. The nice point would be order should be changed with buttons on demand by several options. The main ones are:

- delivered times
- category
- name

**Comment by yajo** [comment]
@rafaelbn with this change, you can already do that in tablets and PC views. You only can't in mobile (because the mobile kanban view doesn't have an ordering option).

**Review by Shide — APPROVED** [review]
Code LGTM

I'm agree with this PR because of that comment:
> @rafaelbn with this change, you can already do that in tablets and PC views. You only can't in mobile (because the mobile kanban view doesn't have an ordering option).

**Comment by rafaelbn** [comment]
Please @yajo could you improve this fields of categories?

imagen

**Comment by yajo** [comment]
@rafaelbn If I do that, it'll affect the way categories are ordered. Are you sure you want that?

**Comment by rafaelbn** [comment]
OK @yajo . Lets, by the moment this as is. I guess this column ins optional (property hide)

**Comment by rafaelbn** [comment]
Don't merge this without @sergio-teruel approval please. This impact in actual implementations in production. Thank you.

**Comment by yajo** [comment]
@fcvalgar was getting a bug fixed in https://github.com/OCA/sale-workflow/pull/2856 while doing functional tests, so I rebased.

**Review by fcvalgar — APPROVED** [review]
Functional review,

LGTM, thak you @yajo

**Review by sergio-teruel — APPROVED** [review]

**Pull request merged**
